x/leveragelp/keeper/msg_server_update_pools.go

```diff
@@ -16,7 +16,11 @@ func (k msgServer) UpdatePools(goCtx context.Context, msg *types.MsgUpdatePools)
 		return nil, errorsmod.Wrapf(govtypes.ErrInvalidSigner, "invalid authority; expected %s, got %s", k.authority, msg.Authority)
 	}
 
-	for _, pool := range msg.Pools {
+	pool, found := k.GetPool(ctx, msg.Pool.AmmPoolId)
+
+	if found {
+		pool.Enabled = msg.Pool.Enabled
+		pool.Closed = msg.Pool.Closed
```

Comment:
I think we don't need whole pool object values when updating, please update it to use required values which are `Enabled` / `Closed` and `PoolId`

Reply:
okay, we don't need `pool` object here